client/src/test/java/io/pravega/client/segment/impl/SegmentOutputStreamTest.java

```diff
@@ -731,7 +730,69 @@ public Void answer(InvocationOnMock invocation) throws Throwable {
         inOrder.verifyNoMoreInteractions();
     }
 
-    
+
+    @Test(timeout = 10000)
+    public void testConnectionFailureWithSegmentSealed() throws Exception {
+        UUID cid = UUID.randomUUID();
+        PravegaNodeUri uri = new PravegaNodeUri("endpoint", SERVICE_PORT);
+        MockConnectionFactoryImpl cf = new MockConnectionFactoryImpl();
+        cf.setExecutor(executorService());
+        MockController controller = new MockController(uri.getEndpoint(), uri.getPort(), cf, true);
+        ClientConnection connection = mock(ClientConnection.class);
+        cf.provideConnection(uri, connection);
+        SegmentOutputStreamImpl output = new SegmentOutputStreamImpl(SEGMENT, true, controller, cf, cid, segmentSealedCallback,
```

Comment:
`@Cleanup` ?

Reply:
it uses a mock connection object, hence did not invoke cleanup here.